crates/services/txpool_v2/src/pending_pool.rs

```diff
@@ -0,0 +1,519 @@
+        missing_inputs: Vec<MissingInput>,
+    ) {
+        let tx_id = transaction.id();
+        self.current_bytes = self
+            .current_bytes
+            .saturating_add(transaction.metered_bytes_size());
+        self.current_gas = self.current_gas.saturating_add(transaction.max_gas());
+        self.current_txs = self.current_txs.saturating_add(1);
+        for input in &missing_inputs {
+            self.unresolved_txs_by_inputs
+                .entry(*input)
+                .or_default()
+                .push(tx_id);
+        }
+        self.unresolved_inputs_by_tx.insert(
+            transaction.id(),
+            ((transaction, insertion_source), missing_inputs),
+        );
+        self.ttl_check.push_front((
+            SystemTime::now()
+                .checked_add(self.ttl)
+                .expect("The system time should be valid; qed"),
+            tx_id,
+        ));
+    }
+
+    fn new_known_utxo(
+        &mut self,
+        utxo_id: UtxoId,
+        resolved_txs: &mut Vec<(ArcPoolTx, InsertionSource)>,
+    ) {
+        if let Some(tx_ids) = self
+            .unresolved_txs_by_inputs
+            .remove(&MissingInput::Utxo(utxo_id))
+        {
+            for tx_id in tx_ids {
+                if let Some((tx, missing_inputs)) =
+                    self.unresolved_inputs_by_tx.remove(&tx_id)
+                {
+                    let missing_inputs: Vec<MissingInput> = missing_inputs
+                        .into_iter()
+                        .filter(|input| match input {
+                            MissingInput::Utxo(utxo) => *utxo != utxo_id,
+                            _ => true,
+                        })
+                        .collect();
+                    if missing_inputs.is_empty() {
+                        resolved_txs.push(tx);
+                    } else {
+                        self.unresolved_inputs_by_tx
+                            .insert(tx_id, (tx, missing_inputs));
+                    }
+                }
+            }
+        }
+    }
+
+    fn new_known_contract(
+        &mut self,
+        contract_id: ContractId,
+        resolved_txs: &mut Vec<(ArcPoolTx, InsertionSource)>,
+    ) {
+        if let Some(tx_ids) = self
+            .unresolved_txs_by_inputs
+            .remove(&MissingInput::Contract(contract_id))
+        {
+            for tx_id in tx_ids {
+                if let Some((tx, missing_inputs)) =
+                    self.unresolved_inputs_by_tx.remove(&tx_id)
+                {
+                    let missing_inputs: Vec<MissingInput> = missing_inputs
+                        .into_iter()
+                        .filter(|input| match input {
+                            MissingInput::Contract(contract) => *contract != contract_id,
+                            _ => true,
+                        })
+                        .collect();
+                    if missing_inputs.is_empty() {
+                        resolved_txs.push(tx);
```

Comment:
You don't clean `self.current_bytes` and related fields when you resolve a transaction and remove it from the pending pool(the same for Utxos).

Reply:
Good catch done.